### ISSUE TITLE
[Bug] Wrong album art showing for music in queue 

### ISSUE THREAD

**Issue by demeralde** [Taiko2k/Tauon#394]
![Screenshot from 2020-10-04 17-07-16](https://user-images.githubusercontent.com/9282837/95008743-20b28a00-0664-11eb-9a2d-6646df048407.png)

**Comment by Taiko2k** [comment]
When you play these tracks in `Tracks + Art` or `Showcase + Lyrics` views, are those the album art as in your screenshot?

**Comment by dspacejs** [comment]
@Taiko2k Yep, it's happening for multiple songs in multiple views. Not just the queue

**Comment by Taiko2k** [comment]
If those tracks are in the same folder, then that's more or less the intended behaviour, since it assumes image files in the same folder are associated with all tracks in that folder. In Tauon the only way to forceably associate an image with a single track is to have it embedded in the file.

**Comment by dspacejs** [comment]
@Taiko2k hmm that's probably why. But all these songs have images embedded in the file, so I'm not sure why it's not working? The embedded image should take priority over files in the same directory.

I'm also confused because the correct album art used to show before, it only broke recently.

**Comment by Taiko2k** [comment]
> I'm also confused because the correct album art used to show before, it only broke recently.

You need to go to `Tracks + Art` view, play a track from the folder then click the album art to cycle it to image 1 (Embedded). The image number selected takes effect per folder.

**Comment by dspacejs** [comment]
@Taiko2k what I mean is the wrong album art shouldn't show at all for a song. It doesn't make sense in any case. It's doing this in multiple places, like songs in the queue, in the right sidebar, and in search results. That's a bug.

**Comment by Taiko2k** [comment]
Tauon doesn't make any assumption about what an image file might be named, If there are incorrect lose album art in the same folder, then that would be regarded as user error.

**Comment by dspacejs** [comment]
@Taiko2k the file itself has the right image. There just seems to be photos in the folders for each song. This is how my music came packaged, going through and reorganising it is a pain (especially since I don't have to with any other media player, they all just use the correct art by default).

It shouldn't be considered user error. Even if it technically was, If a song has album art, that should take priority over any other photo in the same folder regardless.

**Comment by dspacejs** [comment]
@Taiko2k this bug is still present, even after deleting all images from my music folder and each song having the correct imaga metadata

**Comment by Taiko2k** [comment]
@dspacejs If you recently removed images, you may need to click MENU > Database > Reset Image Cache.

Basically, if you start playing a track (or for albums, the first track in the album) you should see its album art in the right side panel. So from there you should be able to figure out where its getting it from. Only that image and the gallery/queue image differ after having pressed "Reset Image Cache" would that be a bug.

**Comment by dspacejs** [comment]
@Taiko2k 

> you may need to click MENU > Database > Reset Image Cach

Didn't work

**Comment by Taiko2k** [comment]
Does the correct image show in the right side panel when playing the track?

**Comment by dspacejs** [comment]
@Taiko2k nope

**Comment by Taiko2k** [comment]
This wrong art must be coming from somewhere then.

When you hover the mouse over the art in the right side panel, you should see an indicator in the bottom right with a letter. **E** for embedded and **F** for file. If its a file, then it must either be in that same folder as the track, or in a subfolder if that subfolders name implies it had album art in it.

**Comment by dspacejs** [comment]
@Taiko2k either way, it should still be prioritising the track's metadata over any images in the folder. It makes sense in terms of priority

**Comment by dspacejs** [comment]
@Taiko2k fixed it. There were images I'd missed because they had a `.` preceding the file name for some reason. I just opened the file browser with hidden files showing and deleted them.

But I still think the priority should be the song's image metadata.

**Comment by C0rn3j** [comment]
I think I agree with priority display of song metadata picture, as you could have an album with an album art them separate songs with their own art.

A problem arises when you have HQ album art as a separate file, and a lower resolution embedded in the files, you'd somehow need to detect that the embedded image isn't just a crappy backup version of the cover you already have in HQ.

**Comment by C0rn3j** [comment]
It does look like song art is actually prioritized over a file already, closing.
